lib/platform/gitlab/index.spec.ts

```diff
@@ -300,6 +301,37 @@ describe(getName(), () => {
       });
       expect(httpMock.getTrace()).toMatchSnapshot();
     });
+    it('should fall back respecting when GITLAB_IGNORE_REPO_URL is set', async () => {
+      process.env.GITLAB_IGNORE_REPO_URL = 'true';
+      const selfHostedUrl = 'http://mycompany.com/gitlab';
+      httpMock
+        .scope(selfHostedUrl)
+        .get('/api/v4/user')
+        .reply(200, {
+          email: 'a@b.com',
+          name: 'Renovate Bot',
+        })
+        .get('/api/v4/version')
+        .reply(200, {
+          version: '13.8',
+        });
+      gitlab.initPlatform({
+        endpoint: `${selfHostedUrl}/api/v4`,
+        token: 'mytoken',
+      });
+      httpMock
+        .scope(selfHostedUrl)
+        .get('/api/v4/projects/some%2Frepo%2Fproject')
+        .reply(200, {
+          default_branch: 'master',
+          http_url_to_repo: `${selfHostedUrl}/some/repo/project.git`,
+        });
+      await gitlab.initRepo({
+        repository: 'some/repo/project',
+        localDir: '',
+      });
+      expect(httpMock.getTrace()).toMatchSnapshot();
```

Comment:
```suggestion
      expect(git.initRepo.mock.calls).toMatchSnapshot();
      expect(httpMock.getTrace()).toMatchSnapshot();
```
so we can match and see if the url is patched the right way